## Mounting the drive

In [102]:
from google.colab import drive
import os
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [159]:
import os
os.chdir('/content/gdrive/My Drive/Datasets/sample_images_7/')
i = 0
for file in os.listdir(os.getcwd()):
  i += 1
  # if 'pokemon' in file:
  # print(file)
    # os.system('rm -rf ' + file)
print(i)

170


In [40]:
os.system('cp new_train_samples.pkl new_train_samples_copy.pkl')

0

## Selecting Files
Since I have two different pickle files(trained at two different time). I load each separately to get all the images from 170 samples

In [156]:
import torch
import pickle as pkl
import matplotlib.pyplot as plt
%matplotlib inline
pkl_file = 'new_train_samples_copy.pkl'    ## change file
img_range = 50 
img_index = 7   ## only change this
folder_name = 'sample_images_' + str(img_index)
head = 0
if 'new' in pkl_file :
  head = 50
  img_range = 120

samples_pkl_in = open('/content/gdrive/My Drive/Datasets/' + pkl_file, 'rb')
samples = pkl.load(samples_pkl_in)
samples_pkl_in.close()

print(head, img_range, len(samples))


50 120 120


In [0]:
import numpy as np
from PIL import Image

**Function to save the files for a each image in a sample.**

In [0]:
def view_samples(epoch, samples, index):
    # fig, axes = plt.subplots(figsize=(16,4), nrows=1, ncols=8, sharey=True, sharex=True)
    img = samples[epoch][img_index]
    img = img.detach().cpu().numpy()
    img = np.transpose(img, (1, 2, 0))
    img = ((img +1)*255 / (2)).astype(np.uint8) # rescale to pixel range (0-255)
    new_img = Image.fromarray(img)
    new_img.save('/content/gdrive/My Drive/Datasets/' + folder_name + '/image_' + str(index+head) + '.jpg')
    # plt.imshow(img)
    # plt.show()
    # for ax, img in zip(axes.flatten(), samples[epoch]):
    #   img = img.detach().cpu().numpy()
    #   img = np.transpose(img, (1, 2, 0))
    #   img = ((img +1)*255 / (2)).astype(np.uint8) # rescale to pixel range (0-255)
    #   # print(min(img), max(img))
    #   ax.xaxis.set_visible(False)
    #   ax.yaxis.set_visible(False)
    #   im = ax.imshow(img.reshape((128,128,3)))
    
    # plt.show()

## Exporting all the Images

In [0]:
for i in range(img_range):
  _ = view_samples(i, samples, i)


## Converting images to video using openCV Video Writer

In [0]:
import cv2
import numpy as np
import glob
 
img_array = []
for filename in glob.glob('/content/gdrive/My Drive/Datasets/sample_images_2/*.jpg'):
    img = cv2.imread(filename)
    height, width, layers = img.shape
    size = (width,height)
    img_array.append(img)
 
 
out = cv2.VideoWriter('/content/gdrive/My Drive/Datasets/project_2.avi',cv2.VideoWriter_fourcc(*'DIVX'), 15, size)
 
for i in range(len(img_array)):
    out.write(img_array[i])
out.release()